src/Aspire.Cli/Templating/DotNetTemplateFactory.cs

```diff
@@ -0,0 +1,249 @@
+            {
+                interactionService.DisplayLines(templateInstallCollector.GetLines());
+                interactionService.DisplayError($"The template installation failed with exit code {templateInstallResult.ExitCode}. For more information run with --debug switch.");
+                return ExitCodeConstants.FailedToInstallTemplates;
+            }
+
+            interactionService.DisplayMessage($"package", $"Using project templates version: {templateInstallResult.TemplateVersion}");
+
+            var newProjectCollector = new OutputCollector();
+            var newProjectExitCode = await interactionService.ShowStatusAsync(
+                ":rocket:  Creating new Aspire project...",
+                async () =>
+                {
+                    var options = new DotNetCliRunnerInvocationOptions()
+                    {
+                        StandardOutputCallback = newProjectCollector.AppendOutput,
+                        StandardErrorCallback = newProjectCollector.AppendOutput,
+                    };
+                    
+                    var result = await runner.NewProjectAsync(
+                                template.Name,
+                                name,
+                                outputPath,
+                                extraArgs,
+                                options,
+                                cancellationToken);
+
+                    return result;
+                });
+
+            if (newProjectExitCode != 0)
+            {
+                interactionService.DisplayLines(newProjectCollector.GetLines());
+                interactionService.DisplayError($"Project creation failed with exit code {newProjectExitCode}. For more information run with --debug switch.");
+                return ExitCodeConstants.FailedToCreateNewProject;
+            }
+
+            await certificateService.EnsureCertificatesTrustedAsync(runner, cancellationToken);
+
+            interactionService.DisplaySuccess($"Project created successfully in {outputPath}.");
+
+            return ExitCodeConstants.Success;
+        }
+        catch (OperationCanceledException)
+        {
+            interactionService.DisplayCancellationMessage();
+            return ExitCodeConstants.FailedToCreateNewProject;
+        }
+        catch (CertificateServiceException ex)
+        {
+            interactionService.DisplayError($"An error occurred while trusting the certificates: {ex.Message}");
+            return ExitCodeConstants.FailedToTrustCertificates;
+        }
+        catch (EmptyChoicesException ex)
+        {
+            interactionService.DisplayError(ex.Message);
+            return ExitCodeConstants.FailedToCreateNewProject;
+        }
+    }
+     
+    private async Task<string> GetProjectNameAsync(ParseResult parseResult, CancellationToken cancellationToken)
+    {
+        if (parseResult.GetValue<string>("--name") is not { } name || !ProjectNameValidator.IsProjectNameValid(name))
```

Comment:
As part of this refactoring we should normalize the project name, not block it.

```suggestion
        if (parseResult.GetValue<string>("--name") is not { } name || !ProjectNameValidator.IsProjectNameValid(name))
```